### DISCUSSION TITLE
Rosbridge: Support reading and setting ROS params

### DISCUSSION THREAD

**Discussion by jtbandes** [foxglove/community#223] [Q&A]
Split from https://github.com/foxglove/studio/issues/1662 

Studio users connected to a data source via a Rosbridge connection (ROS 1 or 2) should be able to:
- Receive and read ROS params from their ROS stack
- Set ROS params by sending values back to the stack

**Comment by bertaveira** [comment]
This would be a very useful feature for when using studio to monitor live a remote device (since native ros2 connection normally has problems opposed to rosbridge). We currently ssh to the device which is not the cleanest or fastest way to change parameters

**Comment by Interpause** [comment]
Modifying node parameters are actually service calls in ROS2. So https://github.com/foxglove/studio/pull/3552 should make it possible to adjust parameters for ROS2. Just waiting for an extension or official panel for service calls now...

**Comment by danclapp4** [comment]
@Interpause @bertaveira Great news, I have written a custom extension to do this exact thing for ros2! with a rosbridge connection, you can view, set, and load ros2 parameters.  I am currently also trying to work on being able to save your parameters to a yml file for if you like your configuration, but want to keep messing with parameters.  

Check it out! https://github.com/danclapp4/ros2-parameter-extension

**Comment by rgov** [comment]
Is there additional context on why the rosbridge data source does not support this yet? It is easy to set a param with roslibjs but perhaps `context.watch("parameters")` would require polling for parameter changes constantly?

**Reply by amacneil** [reply]
I can't speak to why rosbridge doesn't already support parameters, but FYI we have seen many performance issues with rosbridge so are currently working on a new [Foxglove ROS Bridge](https://github.com/foxglove/ros-foxglove-bridge) node written in C++. This will eventually become the primary way to connect to ROS 1 & 2, and fully support parameters, services, and actions.
